docker/docker.go

```diff
@@ -40,6 +39,40 @@ type ContainerInfo struct {
 	UDPPorts    UDPPorts
 }
 
+type EnvVariables map[string]string
+
+// NewDockerClient creates an instance of DockerClient.
+// Returns error when unable to connect to docker daemon.
+func NewDockerClient() (DockerClient, error) {
+	client, err := docker.NewClient(Endpoint)
+	if err != nil {
+		return &ConcreteDockerClient{}, err
+	}
+	return &ConcreteDockerClient{client: client}, nil
+}
+
+// GetRunningContainers finds running containers and returns specific details.
+// TOOD package should return error, logging should be disabled.
+func GetRunningContainers(c DockerClient) ([]ContainerInfo, error) {
+	containersIDs, err := getContainersIDs(c)
+	if err != nil {
+		return []ContainerInfo{}, err
+	}
+	containersDetails, err := getContainersDetails(c, containersIDs)
+	if err != nil {
+		return []ContainerInfo{}, err
+	}
+	return containersDetails, nil
+}
+
+func (c *ConcreteDockerClient) listContainers(opts docker.ListContainersOptions) ([]docker.APIContainers, error) {
+	return c.client.ListContainers(opts)
+}
```

Comment:
It looks a bit strange that we just need to proxy this method. I have a feeling that we're overcomplicating the stuff. As I understand we're just covering the current code with tests, so I'd propose again to cover it with integration tests (stub docker) and then refactor. Applying DI for current code and doing just unit testing could happen to be a longer path.